File: frontend/types/incident.ts
export interface Incident {
  id: string;
  junction_id: string;
  incident_type: string;
  severity: string;
  timestamp: string;
  closed_datetime?: string | null;
  resolved_datetime?: string | null;
  weather: string;
  temperature_c: number;
  description: string;
}

File: backend/app/schemas/incident.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class IncidentResponse(BaseModel):
    id: str
    junction_id: str
    incident_type: str
    severity: str
    timestamp: datetime
    closed_datetime: datetime | None = None
    resolved_datetime: datetime | None = None
    weather: str
    temperature_c: float
    description: str


class IncidentCreateRequest(BaseModel):
    junction_id: str
    incident_type: str
    severity: str
    description: str = ""
    weather: str = "clear"
    temperature_c: float = 25.0
